addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -98,18 +99,33 @@ def _get_bom(self, bom_id=False, product_id=False, line_qty=False, line_id=False
             'bom_prod_name': product.display_name,
             'currency': self.env.user.company_id.currency_id,
             'product': product,
+            'code': bom.code,
             'price': product.uom_id._compute_price(product.standard_price, bom.product_uom_id) * bom_quantity,
             'total': sum([op['total'] for op in operations]),
             'level': level or 0,
             'operations': operations,
             'operations_cost': sum([op['total'] for op in operations]),
+            'attachments': self.env['mrp.document'].search(['|',
+            '&', ('res_model', '=', 'product.product'), ('res_id', '=', product.id), '&', ('res_model', '=', 'product.template'), ('res_id', '=', product.product_tmpl_id.id)]),
             'operations_time': sum([op['duration_expected'] for op in operations])
         }
         components, total = self._get_bom_lines(bom, bom_quantity, product, line_id, level)
         lines['components'] = components
         lines['total'] += total
         return lines
 
+    def _has_attachments(self, bom):
+        has_attachments = False
+        for line in bom.bom_line_ids:
+            if line.has_attachments:
+                has_attachments = True
+                break;
+            if line.child_bom_id:
+                has_attachments = self._has_attachments(line.child_bom_id)
+                if has_attachments:
+                    break;
```

Comment:
maybe make the some logic form bom line and sub bom lines
maybe this should be an helper on the mrp bom model?